### DISCUSSION TITLE
construct token-claim

### DISCUSSION THREAD

**Discussion by quentinalbertone** [skupperproject/skupper#760] [Q&A] [answered]
Hi,

I'm using skupper `0.8.6` and I want to create skupper claim-token thanks to an API,
here is a draft of the function which will create my token-claim and 
I was wondering if the following annotations are required:
* `skuppertypes.SiteVersion`
* `skuppertypes.TokenGeneratedBy`

```Go
claim := v1.Secret{
		TypeMeta: metav1.TypeMeta{
			APIVersion: "v1",
			Kind:       "Secret",
		},
		ObjectMeta: metav1.ObjectMeta{
			Name: name,
			Labels: map[string]string{
                                skuppertypes.SkupperTypeQualifier: skuppertypes.TypeClaimRequest,
				},
			Annotations: map[string]string{
				skuppertypes.ClaimUrlAnnotationKey: skupperEndpoint,
				skuppertypes.SiteVersion:                    s.config.GetSiteMetadata().Version,
				skuppertypes.TokenGeneratedBy:       generatedby,
			},
		},
		Data: map[string][]byte{
			skuppertypes.ClaimPasswordDataKey: password,
			skuppertypes.ClaimCaCertDataKey:   caSecret.Data["tls.crt"],
		},
	}
```

**Comment by quentinalbertone** [comment]
Same question for the v1.0.0 (I have just checked the new release)
Are these annotations used for the `token-claim-record`
```
skupper.io/claim-expiration
skupper.io/claims-remaining
```
Especialy the claim-remaining which wasn't used for the 0.8.6

**Comment by fgiorgetti** [comment, marked as answer]
If you use the Skupper's site-controller ([doc](https://skupper.io/docs/declarative/tutorial.html)) to initialize your
sites, you can simply create a connection token request (as shown below) and Skupper will populate the secret,
so you can download and use it to link remote sites.

```
apiVersion: v1
kind: Secret
metadata:
  labels:
    skupper.io/type: connection-token-request
  name: token-request-1
```

**Reply by quentinalbertone** [reply]
Thanks for your answer, I know that I can create a connection-token-request but if I remember correctly, in the version `0.8.6`, the token populated doesn't have a password and the `skupper.io/claims-remaining`(the claims-remaining wasn't used but I like the idea that I can set a threshold on a claim)
But it could be wrong now.
